src/hardware/mouse/mouse_manymouse.cpp

```diff
@@ -290,6 +294,24 @@ bool ManyMouseGlue::ProbeForMapping(uint8_t &device_id)
 	return success;
 }
 
+bool ManyMouseGlue::IsCancelRequested()
```

Comment:
Looks like a nicely packaged stdin checker for many of our other interactive CLI tools that should/could be user-cancellable. 

Things like `more`, or a long pipe operation, (maybe others?)

If so, maybe it could be moved to dos.cpp (misc, ioctcl, or files).

Reply:
The problem is that each command has quite a different expectations from waiting code:

- MOUSECTL reacts on CTRL+C/ ESC / Q (to terminate operation), and it needs a waiting function to return if user does not press anything (contrary to the others)
- PAUSE reacts on any key
- CHOICE reacts on one of the supplied keys, or on CTRL+C to terminate
- MORE reacts on CTRL+C / ESC / Q (to terminate operation), on SPACE / ENTER (to display more rows) and on N (to go to the next file - but only if more than one file was supplied)

I'll try to write some solution, but it won't be one trivial function.

Reply:
Added choice functions suitable for MORE and MOUSECTL, in a separate commit. Still to do: CHOICE and PAUSE.

Reply:
I have added generic functions usable for MOUSECTL and MORE - see the new commit.
CHOICE is very specific, and PAUSE is really trivial - so I prefer to leave them as is.